src/devices_models/device_constructors/common/constructor_validations.jl

```diff
@@ -12,58 +12,53 @@ function validate_available_devices(
     return true
 end
 
-function validate_services!(
-    service::Type{S},
-    services::Vector{S},
+function validate_service!(
+    model::ServiceModel{S, <:AbstractServiceFormulation},
     incompatible_device_types::Vector{<:DataType},
     sys::PSY.System,
 ) where {S <: PSY.Service}
-    services_ = PSY.get_components(S, sys)
-    if isempty(services_)
-        @warn "The data doesn't include services of type $(service), consider changing the service models" _group =
+    service = PSY.get_component(S, sys, get_service_name(model))
+    if isnothing(service)
+        @warn "The data doesn't include services of type $(S) and name $(get_service_name(model)), consider changing the service models" _group =
             :ConstructGroup
         return false
     end
 
     services_mapping = PSY.get_contributing_device_mapping(sys)
-    for s in services_
-        contributing_devices_ =
-            services_mapping[(type = S, name = PSY.get_name(s))].contributing_devices
-        contributing_devices = [
-            d for d in contributing_devices_ if
-            typeof(d) ∉ incompatible_device_types && PSY.get_available(d)
-        ]
-        if isempty(contributing_devices)
-            @warn "The contributing devices for service $(PSY.get_name(s)) is empty, consider removing the service from the system" _group =
-                :ConstructGroup
-        else
-            push!(services, s)
-        end
+
+    contributing_devices_ =
```

Comment:
I wonder if we should be caching this calculation somewhere else. But we can improve this later.

Reply:
I'll add a TODO here so we can address this later